constraint/pkg/apis/templates/transform_test.go

```diff
@@ -0,0 +1,330 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+func TestAddPreserveUnknownFields(t *testing.T) {
+	trueBool := true
+	falseBool := false
+	testCases := []struct {
+		name  string
+		v     *apiextensionsv1beta1.JSONSchemaProps
+		exp   *apiextensionsv1beta1.JSONSchemaProps
+		error bool
+	}{
+		{
+			name: "no information",
+			v:    &apiextensionsv1beta1.JSONSchemaProps{},
+			exp: &apiextensionsv1beta1.JSONSchemaProps{
+				XPreserveUnknownFields: &trueBool,
+			},
+		},
+		{
+			name: "nil properties",
+			v: &apiextensionsv1beta1.JSONSchemaProps{
+				Properties: nil,
+			},
+			exp: &apiextensionsv1beta1.JSONSchemaProps{
+				XPreserveUnknownFields: &trueBool,
+				Properties:             nil,
+			},
+		},
+		{
+			name: "x-kubernetes-preserve-unknown-fields is present already but false",
+			v: &apiextensionsv1beta1.JSONSchemaProps{
+				XPreserveUnknownFields: &falseBool,
```

Comment:
`false` appears to not be invalid for this field.

https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types_jsonschema.go#L95

https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1beta1/types_jsonschema.go#L95

Reply:
Ah, nice.  We can take this out @maxsmythe